### PR TITLE
feat(streaming): enable lookup to scan from arrange keyspace

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

## What's changed and what's your intention?

Refactor meta to allow lookup executor to scan from arrangements. We store a fragment_id in lookup stream node.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)

close https://github.com/singularity-data/risingwave/issues/719

### CODE REVIEW

**Comment by skyzh** [comment]
Also create delta-join mv is extremely slow (5 secs), not sure what's happening... I'll look into this later.

File: src/meta/src/stream/rewrite/delta_join.rs [hunk 1]
```diff
@@ -213,6 +213,9 @@ where
                 arrange_key: hash_join_node.left_key.clone(),
                 stream_key: hash_join_node.right_key.clone(),
                 use_current_epoch: false,
+                arrange_fragment_id: u32::MAX,
```
Comment (BugenZhao) [inline src/meta/src/stream/rewrite/delta_join.rs:216]: Is this about to be rewritten? If so, please add some comments here.

File: src/stream/src/executor_v2/lookup.rs [hunk 2]
```diff
@@ -117,11 +125,22 @@ impl ExecutorBuilder for LookupExecutorBuilder {
...
             .collect();
 
+        assert_ne!(
+            node.arrange_fragment_id,
+            u32::MAX,
+            "arrange fragment id is not available"
+        );
+        let arrangement_keyspace_id =
+            unique_operator_id(node.arrange_fragment_id, node.arrange_operator_id);
+
```
Comment (yuhao-su) [inline src/stream/src/executor_v2/lookup.rs:135]: Perhaps make the calculation of arrangement keyspace a function of `ArrangeExecutor`? Can do it in later PRs.

**Comment by yuhao-su** [comment]
LGTM!

**Review by fuyufjh — APPROVED** [review]
LGTM

File: src/stream/src/executor_v2/lookup.rs (hunk 2 above)
Comment (skyzh) [reply src/stream/src/executor_v2/lookup.rs:135]: We will move toward using `table_id` in the future. Also we don't have `ArrangeExecutor`. We are simply using `MaterializeExecutor` for storing data.

**Pull request merged**